API/Sync mode (#4)
changebase-io/ruby-gem: README.md

### Configuration

#### Replication Mode

> The default model for the `changebase` gem is `replication`. In this mode

Reviewer: ```suggestion
The default mode for the `changebase` gem is `replication`. In this mode
```

Changebase is setup to replicate your database and record events via the
replication stream.

> By default without any configuration `changebase` will write metadata to the
> `"changebase_metadata"`. To configure the metadata table that Changebase writes
> to create a initializer at `config/initializers/changebase.rb` with the following:

Reviewer: ```suggestion
The default configuration will write metadata to the `"changebase_metadata"` table. To
configure the metadata table, create an initializer at `config/initializers/changebase.rb`
with the following:
```

Changebase.metadata_table = "my_very_cool_custom_metadata_table"
```

#### API/Sync Mode

> In the event you cannot setup your database to replicate to changebase you can
> use the synchronous API mode. Events will be sent to Changebase over our HTTP API.
> This mode will enable you collect roughly the same information but has the
> potentional to miss events and changes in your database if you are not careful.

Reviewer: ```suggestion
If you are unable to setup database replication you can use the synchronous API
mode. Events will be sent to Changebase through the API. This mode will enable
you collect roughly the same information but has the potentional to miss events
and changes in your database if you are not careful, or if another application
accesses the database directly.
```

To configure Changebase in the `"api/sync"` mode create a initializer at
`config/initializers/changebase.rb` with the following:

```ruby
Rails.application.config.tap do |config|
  config.changebase.mode = "api/sync"

>   config.changebase.connection = "https://API_KEY@chanbase.io"

Reviewer: ```suggestion
  config.changebase.connection = "https://#{ ENV.fetch('CHANGEBASE_API_KEY') }@changebase.io"
```
